pmm-app/src/shared/components/Elements/Charts/Sparkline/Sparkline.tsx

```diff
@@ -1,177 +1,221 @@
+      const columnNumber = Math.floor(e.offsetX / (GRAPH_WIDTH / (drawData.length - 1)));
+
+      setTooltip(createTooltip(columnNumber));
 
-      setTooltip(dataTooltip);
+      updateGraphs(columnNumber);
+
+      // ReactTooltip.show(sparklineCanvas);
+      ctx.clearRect(0, 0, 300, 30);
+      drawData.forEach((item, index) => {
+        drawBar(index, color);
+      });
+
+      drawBar(columnNumber, 'white', true);
     });
-    svg.on('mouseover', () => focusG.style('display', null));
-    svg.on('mouseout', () => focusG.style('display', 'none'));
 
-    // Create X axis
-    const xAxis = axisBottom(scaleX);
+    sparklineCanvas.current.addEventListener('mouseout', () => {
+      updateGraphs(NaN);
+      ctx.clearRect(0, 0, 300, 30);
+      drawData.forEach((item, index) => {
+        drawBar(index, color);
+      });
+    });
 
-    svg
-      .append('g')
-      .attr('class', 'x-axis')
-      .attr('transform', `translate(${margin},${height - margin - 1})`)
-      .call(xAxis);
-  };
+    // sparklineCanvas.current.addEventListener('mousedown', (e) => {
+    //   const startPoint = e.offsetX;
+    //   let endPoint = null;
+    //
+    //   e.stopPropagation();
+    //   const mouseUpHandler = (e) => {
+    //     ctx.clearRect(startPoint, 0, endPoint - startPoint, BAR_HEIGHT);
+    //
+    //     const start = Math.floor(startPoint / (GRAPH_WIDTH / drawData.length));
+    //     const end = Math.floor(endPoint / (GRAPH_WIDTH / drawData.length));
+    //
+    //     const result = [appLoadPolygonChart[start], appLoadPolygonChart[end]]
+    //       .sort((a, b) => b.point - a.point)
+    //       .map((item) => item.timestamp);
+    //
+    //     console.log(result);
+    //     e.stopPropagation();
+    //     sparklineCanvas.current.removeEventListener('mouseup', mouseUpHandler);
+    //     sparklineCanvas.current.removeEventListener('mouseup', mouseMove);
+    //   };
+    //   const mouseMove = (e) => {
+    //     endPoint = e.offsetX;
+    //
+    //     // ctx.fillStyle = "rgba(150,150,150,0.3)";
+    //     // const height = drawData[barIndex].y;
+    //     // ctx.fillRect(startPoint, 0, endPoint - startPoint, BAR_HEIGHT);
+    //
+    //     e.stopPropagation();
+    //   };
+    //
+    //   sparklineCanvas.current.addEventListener('mouseup', mouseUpHandler);
+    //   sparklineCanvas.current.addEventListener('mousemove', mouseMove);
+    // });
+    ctx.clearRect(0, 0, 300, 30);
+    drawData.forEach((item, index) => {
+      drawBar(index, color);
+    });
+
+    const drawHighlighted = (e) => {
+      ctx.clearRect(0, 0, 300, 30);
+      drawData.forEach((item, index) => drawBar(index, color));
+
+      if (!e.detail) {
+        return;
+      }
+
+      const columnNumber = e.detail;
+
+      drawBar(columnNumber, '#00e6e6');
+    };
+
+    document.addEventListener('sync-graphs', drawHighlighted, false);
+
+    // eslint-disable-next-line consistent-return
+    return () => {
```

Comment:
Does it work without the curly braces? (e.g. `() => document.removeEventListener('sync-graphs', drawHighlighted);`)